### PR TITLE
Focus trap & dialog component

### PR DESCRIPTION
**Pull request by corinagum**
- add dialog component for use in appSettings, aboutDialog, & conversationsSettingsDialog
- add focus trap on dialog
- some styling fixes so that focus is visible on elements within dialog
- changed standard blue to constrast compliant one

### CODE REVIEW

**Comment by eanders-MS** [comment]
About Dialog: When opened, the aka.ms link is focused. Was that intentional? I'd prefer the [x] button be focused.

**Comment by eanders-MS** [comment]
Could you please add: Escape key should close the dialog. Thanks.

**Comment by corinagum** [comment]
@eanders-MS Woops, I somehow forgot to do the escape, thanks for catching. 
For the Dialog auto-focus: the link being focused was intentional, but I'm happy to change it. Does this mean that all three dialogs should start focus on the [x]button?

**Comment by eanders-MS** [comment]
Let's focus the first editable UI component if one exists (edit box, checkbox, etc... something the user can modify).

**Comment by corinagum** [comment]
sounds good.

File: src/client/dialogs/appSettingsDialog.tsx [hunk 1]
```diff
@@ -120,17 +111,24 @@ export class AppSettingsDialog extends React.Component<{}, AppSettingsDialogStat
     }
 
     componentWillUnmount() {
-        window.removeEventListener('click', this.pageClicked);
         this.settingsUnsubscribe();
     }
 
+    onKeyUpEnterNav(event, name) {
```
Comment (compulim) [inline src/client/dialogs/appSettingsDialog.tsx]: How about renaming this to `handleKeyUpEnterNav`, or `handleKeyUp`?

**Comment by corinagum** [comment]
@eanders-MS @compulim not sure if you get updates when I fix PRs, but I just added escape & fixed the focus. 👍

**Comment by eanders-MS** [comment]
Getting closer! Something knocked the copyright notice off-center:
image

**Comment by corinagum** [comment]
@eanders woopsies here you go

**Comment by eanders-MS** [comment]
Looks good. Thank you @corinagum !

**Pull request merged**